File: dart/selection_sort.dart
List selectedSort(List arr) {
  for (int i = 0; i < arr.length - 1; i++) {
    int min_val_index = i;
    for (int j = i + 1; j < arr.length; j++) {
      if (arr[j] < arr[min_val_index]) {
        min_val_index = j;
      }
    }
    if (min_val_index != i) {
      var temp = arr[i];
      arr[i] = arr[min_val_index];
      arr[min_val_index] = temp;
    }
  }
  return arr;
}

File: c#/selection_sort.cs
public static int[] SelectionSort(int[] arr)
{
  for (int i = 0; i < arr.Length; i++)
  {
    int min_value_index = i;
    for (int j = i; j < arr.Length; j++)
    {
      if (arr[j] < arr[min_value_index])
      {
        min_value_index = j;
      }
    }
    if (min_value_index != i)
    {
      int temp = arr[i];
      arr[i] = arr[min_value_index];
      arr[min_value_index] = temp;
    }
  }
  return arr;
}